sdk/identity/identity/src/extensionProvider.ts

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation
+// Licensed under the MIT license.
+
+/**
+ * ===== Warning! =====
+ *
+ * It's important that this module does not contain any imports or exports.
+ *
+ * This file is referenced by the other identity packages, such as
+ * identity-vscode and identity-persistence, so any imports or exports here will
+ * be dragged into those packages' depdendencies as well.
+ */
+
+// #region Global Extension Registry
+
+/**
+ * About this functionality:
+ *
+ * We needed a way to allow Identity packages to pass implementation objects
+ * back and forth without adding those items to the public API surface of
+ * the base identity package, so we are using a symbol-property of the global
+ * object to store a map from symbols to implementations.
+ *
+ * This allows extension _providers_ to reference this file directly and to
+ * register their extensions with the extension system while exporting only
+ * a symbol from their package API. This allows those packages to utilize
+ * MSAL/Identity internals without exposing those internals.
+ */
+
+export type AzureExtensionRegistry = Map<symbol, (context: AzureExtensionContext) => void>;
+
+const registryKey: unique symbol = Symbol.for("__azure_identity_extensions");
+
+const _glb: { [registryKey]: AzureExtensionRegistry } = (globalThis ?? global ?? window) as any;
+
+export const registry: Map<symbol, (context: AzureExtensionContext) => void> =
```

Comment:
I'm a little confused why `useIdentityExtension` can't handle the registration instead of the extension package needing to reach into a global registry object?

Reply:
Removed as part of the extension APi rework.